interfaces/Matlab_Toolbox_Revamp/Class/Kinetics.m

```diff
@@ -0,0 +1,590 @@
+classdef Kinetics < handle
+    
+    properties
+        owner
+        kin_id
+        Kc % equilibrium constant
+        Kf % forward reaction rate
+        Kr % reverse reaction rate
+        dH % enthalpy of reaction
+        dS % entropy of reaction
+        dG % gibbs free energy of reaction
+    end
+
+    methods
+        %% Kinetics class constructor
+
+        function kin = Kinetics(ph, src, id, n1, n2, n3, n4)
+            checklib;
+            % indices for bulk phases in a heterogeneous mechanism
+            inb1 = -1;
+            inb2 = -1;
+            inb3 = -1;
+            inb4 = -1;
+            if nargin == 2
+                id = '-';
+            end
+            kin.owner = 1;
+            % get the integer indices used to find the stored objects
+            % representing the phases participating in the mechanism
+            iph = ph.tp_id;
+            if nargin > 3
+                inb1 = n1.tp_id;
+                if nargin > 4
+                    inb2 = n2.tp_id;
+                    if nargin > 5
+                        inb3 = n3.tp_id;
+                        if nargin > 6
+                            inb4 = n4.tp_id;
+                        end
+                    end
+                end
+            end
+            kin.kin_id = calllib(ct, 'kin_newFromFile', src, id, ...
+                                 iph, inb1, inb2, inb3, inb4);
+        end
+        
+        %% Utility methods
+
+        function kin_clear(kin)
+            % Delete the kernel object
+            
+            checklib;
+            calllib(ct, 'kin_del', kin.kin_id);
+        end
+        
+        %% Get scalar attributes
+
+        function n = isReversible(kin, i)
+            % Get an array of flags indicating reversibility of a reaction.
+            %
+            % parameter i:
+            %    Integer reaction number.
+            % return:
+            %    1 if reaction number i is reversible. 0 if irreversible.
+            
+            checklib;
+            n = calllib(ct, 'kin_isReversible', kin.kin_id, i);
+        end
+        
+        function n = multiplier(kin, irxn)
+            % Get the multiplier for reaction rate of progress.
+            %
+            % parameter irxn:
+            %    Integer reaction number for which the multiplier is
+            %    desired.
+            % return:
+            %    Multiplier of the rate of progress of reaction irxn.
+            
+            checklib;
+            n = calllib(ct, 'kin_multiplier', kin.kin_id, irxn-1);
+        end
+        
+        function n = nReactions(kin)
+            % Get the number of reactions.
+            %
+            % return:
+            %    Integer number of reactions
+            
+            checklib;
+            n = calllib(ct, 'kin_nReactions', kin.kin_id);
+        end
+        
+        function n = nSpecies2(kin)
```

Comment:
Why does this have a `2` suffix on the method name?

Reply:
There is already a method called `nSpecies` in the `ThermoPhase` class. I renamed it to avoid conflicts since the `Solution` class inherits from both `ThermoPhase` and `Kinetics`.

Reply:
What does the Python interface do in this case? Does it have a similar problem, and how is it resolved?